runtime/lua/vim/render.lua

```diff
@@ -0,0 +1,162 @@
+-- Extra functionality for rendering rich text buffers (vim help, markdown)
+
+local M = {}
+
+--- Apply current colorscheme to lists of default highlight groups
+---
+--- Note: {patterns} is assumed to be sorted by occurrence in the file.
+--- @param patterns {start:string,stop:string,match:string}[]
+function M.highlight_groups(patterns)
```

Comment:
should this live in `vim.hl` ?

```suggestion
function M.hl_groups(patterns)
```

Reply:
"Highlight" is the verb here :) So it should be `highlight_hl_groups`.